### PR TITLE
jsonschemas: add 2 new document types

### LINKED ISSUE #626: Problems in the document editor within field `partOf`
**Describe the issue**

In the document editor, with the partOf field
1. Sometimes I cannot to add the following sub-fields: `numberingVolume`, `numberingIssues` depending of the document type.
2. When editing an existing document, the sub-field `document.contribution` is not displayed even if it already contains data

This might be the case also for other sub-fields.

**To Reproduce**
1. Open a new document
2. Add field "Contained in (host document" (`partOf`)
3. Try to add the mentioned subfields

**Expected behavior**
1. Allow to add
    * `numberingVolume` (document) and `volume` (deposit) for the following doc types: coar:c_3248, coar:c_3e5a, coar:c_beb9, coar:c_6501, coar:c_998f, coar:c_dcae04bc, coar:c_816b, coar:c_5794, coar:c_6670, coar:c_c94f
    * `numberingIssue` (document) and `issue` (deposit) for the following doc types: coar:c_3e5a, coar:c_beb9, coar:c_6501, coar:c_998f, coar:c_dcae04bc, coar:c_816b, coar:c_5794, coar:c_6670, coar:c_c94f
    * Always allow to add `numberingPages` / `pages` and `numberingYear` / `year`
1. Solve problem 2

**Software version**
sonardev.test.rero.ch AND sonar.test.rero.ch

**Screenshots**
image

### LINKED ISSUE #881: Add 2 document types: conference object, book review
### How it works

1. On request of the SONAR users, two document types are needed:
* `coar:R60J-J5BD` - Conference presentation
* `coar:c_ba08` - Book review

2. Other thesis is not clear in the JSON schema.

### Improvement suggestion

1. Add this 2 document types
* in the JSON schema of the deposit and of the document
    * `coar:R60J-J5BD` (Conference presentation): settings similar to `coar:c_18co` (conference poster not in proceedings)
    * `coar:c_ba08` (Book review): settings similar to `coar:c_3e5a` (contribution to a journal)
* in the [schema.org](https://github.com/rero/sonar/blob/staging/sonar/modules/documents/serializers/schemas/schemaorg.py) mapping:
    * `coar:R60J-J5BD` = CreativeWork
    * `coar:c_ba08` = Review

2. Rename "Other" (under "Thesis") to "Other type of thesis"
    - replace `document_type_other` by `document_type_other_thesis`
    - check if `other` is used in hide conditions

### PR DESCRIPTION
**Pull request by PascalRepond**
* Adds conference object and book review document types for documents
and deposits.
* Changes value and label for document_type_other for better clarity.
* Initializes alembic.
* Includes alembic script for db migration.
* Closes #881.
* Closes #626.
* Makes the `status` deposit field required to prevent submitting
a deposit with no status from the API.

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

**Review by jma — APPROVED** [review]

File: sonar/modules/documents/serializers/schemas/schemaorg.py [hunk 1]
```diff
@@ -71,7 +71,9 @@
     'advanced_studies_thesis': 'Thesis',
     'other': 'Thesis',
```
Comment (pronguen) [inline sonar/modules/documents/serializers/schemas/schemaorg.py:72]:
```suggestion
    'other_thesis': 'Thesis',
```

**Review by pronguen — CHANGES_REQUESTED** [review]
2 problems regarding https://github.com/rero/sonar/issues/881

File: sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json [hunk 2]
```diff
@@ -573,7 +574,7 @@
               "type": "string",
               "minLength": 1,
               "form": {
-                "hideExpression": "field.parent.parent.model && !['coar:c_beb9', 'coar:c_6501', 'coar:c_998f', 'coar:c_dcae04bc'].includes(field.parent.parent.model.documentType)",
+                "hideExpression": "field.parent.parent.model && !['coar:c_3248', 'coar:c_3e5a', 'coar:c_ba08', coar:c_beb9', 'coar:c_6501', 'coar:c_998f', 'coar:c_dcae04bc', 'coar:c_816b', 'coar:c_5794', 'coar:c_6670', 'coar:c_c94f'].includes(field.parent.parent.model.documentType)",
```
[the remark this answers is not in the document]
Comment (PascalRepond) [reply sonar/modules/deposits/jsonschemas/deposits/deposit-v1.0.0_src.json:577]: Oops, bien vu!

File: sonar/modules/documents/serializers/schemas/schemaorg.py (hunk 1 above)
Comment (PascalRepond) [reply sonar/modules/documents/serializers/schemas/schemaorg.py:72]: Re-bien vu! Merci

**Comment by pronguen** [comment]
*[work in progress...]*

Problems regarding https://github.com/rero/sonar/issues/626

- [ ] Field "numberingVolume" (document) still not available for the specified document types
image

**Pull request merged**